tests/run-test-batches.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
```

Comment:
Please add a documentation comment to the top of the file with the different allowed env variables and arguments.

I would also say to move all the globals and the functions above, mix of scripting/functions defs/globals makes it a bit harder to read, especially as this is a useful script to use ourselves too, not just for the CI